tests/themis/themis_secure_cell.c

```diff
@@ -32,6 +32,35 @@ static char passwd[] = "password";
 static char message[] = "secure cell test message by Mnatsakanov Andrey from Cossack Labs";
 static char user_context[] = "secure cell user context";
 
+static void secure_cell_key_generation(void)
+{
+    themis_status_t res;
+    uint8_t master_key[MAX_KEY_SIZE];
+    size_t master_key_length = 0;
+
+    res = themis_gen_sym_key(NULL, &master_key_length);
+    testsuite_fail_unless(res == THEMIS_BUFFER_TOO_SMALL, "keygen: query key size");
+
+    res = themis_gen_sym_key(master_key, &master_key_length);
+    testsuite_fail_unless(res == THEMIS_SUCCESS, "keygen: generate new key");
+    testsuite_fail_unless(master_key_length > 0, "keygen: key length returned");
```

Comment:
will be better to more precisely compare result key length and check that `master_key_length == MAX_KEY_SIZE`

plus in tests of key generation, we should check that our generation function did something. for example compare that values of an array were changed after generation and not equal to previous values. plus we should check that each call of function return new random values and user can call this function as many times as he want to get new key, without calling somehing like `RAND_seed(...)` in openssl. So we should check at least 2 calls and verify that it return different values

Reply:
> we should check that each call of function return new random values

Sounds fair. It seems a good idea to test whether the cryptographic backend is sane. Standby for more tests.